cmd/jaeger/internal/exporters/storageexporter/exporter.go

```diff
@@ -30,14 +35,22 @@ func newExporter(config *Config, otel component.TelemetrySettings) *storageExpor
 	}
 }
 
-func (exp *storageExporter) start(_ context.Context, host component.Host) error {
+func (exp *storageExporter) start(ctx context.Context, host component.Host) error {
 	f, err := jaegerstorage.GetStorageFactory(exp.config.TraceStorage, host)
 	if err != nil {
 		return fmt.Errorf("cannot find storage factory: %w", err)
 	}
 
-	if exp.spanWriter, err = f.CreateSpanWriter(); err != nil {
-		return fmt.Errorf("cannot create span writer: %w", err)
+	switch t := f.(type) {
+	case *ch.Factory:
+		exp.clickhouse = true
+		t.CreateSpansTable(ctx)
```

Comment:
why can't we do this inside `clickhouse.NewFactory`? `storageexporter` doesn't need to know about internal details of each storage implementation